src/operator/controllers/protected_services_reconcilers/default_deny.go

```diff
@@ -0,0 +1,171 @@
+package protected_services_reconcilers
+
+import (
+	"context"
+	"fmt"
+	otterizev1alpha2 "github.com/otterize/intents-operator/src/operator/api/v1alpha2"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/otterize/intents-operator/src/shared/operatorconfig"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	v1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DefaultDenyReconciler reconciles a ProtectedServices object
+type DefaultDenyReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	injectablerecorder.InjectableRecorder
+}
+
+func NewDefaultDenyReconciler(client client.Client, scheme *runtime.Scheme) *DefaultDenyReconciler {
+	return &DefaultDenyReconciler{
+		Client: client,
+		Scheme: scheme,
+	}
+}
+
+func (r *DefaultDenyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if !viper.GetBool(operatorconfig.EnableProtectedServicesKey) {
+		return r.DeleteAllDefaultDeny(ctx, req.Namespace)
+	}
+
+	var ProtectedServicesResources otterizev1alpha2.ProtectedServicesList
+
+	err := r.List(ctx, &ProtectedServicesResources, client.InNamespace(req.Namespace))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.blockAccessToServices(ctx, ProtectedServicesResources, req.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *DefaultDenyReconciler) blockAccessToServices(ctx context.Context, ProtectedServicesResources otterizev1alpha2.ProtectedServicesList, namespace string) error {
+	serversToProtect := map[string]v1.NetworkPolicy{}
+	for _, list := range ProtectedServicesResources.Items {
+		if list.DeletionTimestamp != nil {
+			logrus.Infof("Protected service %s is being deleted and ignored", list.Name)
```

Comment:
This doesn't actually ignore the protected service - it just doesn't add it to the list of servers to protect for the calculation later. This message confused me since I was wondering how it then gets deleted.

But I think you can remove the log altogether - this is part of normal functioning, shouldn't have to log this. What do you think?

Reply:
This is all an already merged code, I just moved it around